core/service/src/chain_ops.rs

```diff
@@ -103,13 +118,18 @@ pub fn import_blocks<F, E, R>(
 ) -> error::Result<()>
 	where F: ServiceFactory, E: Future<Item=(),Error=()> + Send + 'static, R: Read,
 {
-	struct DummyLink;
-	impl<B: Block> Link<B> for DummyLink { }
-
 	let client = new_client::<F>(&config)?;
 	// FIXME #1134 this shouldn't need a mutable config.
 	let queue = components::FullComponents::<F>::build_import_queue(&mut config, client.clone())?;
-	queue.start(DummyLink)?;
+
+	let (wait_send, wait_recv) = std::sync::mpsc::channel();
+	impl<B: Block> Link<B> for WaitLink {
+		fn block_imported(&self, _hash: &B::Hash, _number: NumberFor<B>) {
+			self.wait_send.send(()).expect("Unable to notify about imported block");
+		}
+	}
```

Comment:
pedantic, but I don't see any reason for this impl being placed here rather than next to the definition. 

Or you can move the definition here. I think in this case you can omit `new` since it would be rather verbose.